app/assets/v2/js/grants/detail.js

```diff
@@ -49,36 +71,34 @@ $(document).ready(function() {
     editableFields.forEach(field => disableEdit(field));
   });
 
-  $('#js-cancel_grant').validate({
-    submitHandler: function(form) {
-      var data = {};
-
-      $.each($(form).serializeArray(), function() {
-        data[this.name] = this.value;
-      });
-
-      // let compiledSubscription;
-      //
-      // if (data.contract_version == 0) {
-      //   compiledSubscription = compiledSubscription0;
-      // }
-
-      let deployedSubscription = new web3.eth.Contract(compiledSubscription.abi, data.contract_address);
-
-      web3.eth.getAccounts(function(err, accounts) {
-        deployedSubscription.methods.endContract()
-          .send({from: accounts[0], gasPrice: 4000000000})
-          .on('transactionHash', function() {
-            document.issueURL = document.getElementById('form--input__reference-url').value;
-            enableWaitState('#grants-details');
-          })
-          .on('confirmation', function(confirmationNumber, receipt) {
-            console.log('receipt', receipt);
-            form.submit();
+  $('#cancel_grant').click(function() {
+
+    let contract_address = $('#contract_address').val();
+
+    let deployedSubscription = new web3.eth.Contract(compiledSubscription.abi, contract_address);
+
+    web3.eth.getAccounts(function(err, accounts) {
+      deployedSubscription.methods.endContract()
+        .send({from: accounts[0], gasPrice: 4000000000})
+        .on('transactionHash', function() {
+          enableWaitState('#grants-details');
+        })
+        .on('confirmation', function(confirmationNumber, receipt) {
+          $.ajax({
+            type: 'post',
+            url: '',
+            data: { 'contract_address': contract_address},
+            success: function(json) {
+              console.log('cancel grant POST successful');
```

Comment:
Maybe a page reload here so the changes will be reflected.

Reply:
yes, the redirects in views.py dont seem to be firing. Aditya had added some but i removed them thinking we didnt need them. I will add back in here and ln55.

Reply:
Yes because is an ajax sent not normal form so you will need something like `location.reload();` in the success callback.
Also to get accepted by stickler you need to remove all the console.log()

Reply:
any suggestions for the failure case besides a console.log?

Reply:
You can probably make an alert rather than a console log. Use the `_alert()` function maybe.